sdks/java/core/src/main/java/org/apache/beam/sdk/io/CountingSource.java

```diff
@@ -83,6 +83,20 @@
   }
 
   /**
+   * Creates a {@link BoundedSource} that will produce the specified number of elements,
+   * from {@code startOffset} to {@code numElements - 1}.
+   *
+   * @deprecated use {@link CountingInput#forSubrange(long, long)} instead
+   */
+  @Deprecated
```

Comment:
Why add a method and immediately deprecate it? If you want users to not use it, package-local visibility is sufficient for that.